crates/ide/src/references/rename.rs

```diff
@@ -659,7 +737,7 @@ mod tests {
     fn test_prepare_rename_namelikes() {
         check_prepare(r"fn name$0<'lifetime>() {}", expect![[r#"3..7: name"#]]);
         check_prepare(r"fn name<'lifetime$0>() {}", expect![[r#"8..17: 'lifetime"#]]);
-        check_prepare(r"fn name<'lifetime>() { name$0(); }", expect![[r#"23..27: name"#]]);
+        check_prepare(r"fn name<'lifetime>() { name$0(); }", expect![[r#"3..7: name"#]]);
```

Comment:
Heh, this was actually a legitimate test failure! Sometimes, `expect` is too easy to update

Reply:
Actually the range for `prepare_rename` was intended to be of the identifier at the cursor position that is going to be renamed. But after rereading the lsp docs I'm not sure if its supposed to be of the identifier under the cursor or the identifier of target definition, https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_prepareRename. The docs aren't as clear as I'd like them to be at times.

Reply:
It should be ident under cursor -- that's for the IDE to highlight it and show "rename" box. Docs are sometimes confusing, but the feature not working in VS Code because of the wrong range is not :D